packages/instantsearch.js/src/widgets/autocomplete/autocomplete.tsx

```diff
@@ -193,31 +198,19 @@ function AutocompleteWrapper<TItem extends BaseHit>({
         }),
     });
 
-  const query =
-    instantSearchInstance.getUiState()[targetIndex!.getIndexId()].query;
-
-  const searchBoxSuit = component('SearchBox');
-  const searchBoxCssClasses = {
-    root: cx(searchBoxSuit()),
-    form: cx(searchBoxSuit({ descendantName: 'form' })),
-    input: cx(searchBoxSuit({ descendantName: 'input' })),
-    submit: cx(searchBoxSuit({ descendantName: 'submit' })),
-    submitIcon: cx(searchBoxSuit({ descendantName: 'submitIcon' })),
-    reset: cx(searchBoxSuit({ descendantName: 'reset' })),
-    resetIcon: cx(searchBoxSuit({ descendantName: 'resetIcon' })),
-    loadingIndicator: cx(searchBoxSuit({ descendantName: 'loadingIndicator' })),
-    loadingIcon: cx(searchBoxSuit({ descendantName: 'loadingIcon' })),
-  } satisfies SearchBoxComponentCSSClasses;
+  const query = isolatedIndex?.getHelper()?.state.query;
 
   return (
     <Autocomplete {...getRootProps()} classNames={cssClasses}>
-      <SearchBox
-        query={query}
-        refine={refine}
-        // @ts-ignore type issues on event handlers
-        inputProps={getInputProps()}
-        cssClasses={searchBoxCssClasses}
-        templates={searchBoxTemplates}
+      <AutocompleteSearchBox
+        query={query || ''}
+        inputProps={{
+          ...getInputProps(),
+          onInput: (evt: JSXPreact.TargetedEvent<HTMLInputElement>) =>
+            refine(evt.currentTarget.value),
```

Comment:
Nitpick: `<AutocompleteSearchBox>` could instead switch to `onInput`: https://preactjs.com/guide/v10/differences-to-react/#use-oninput-instead-of-onchange.